### PR TITLE
Adopt react-bootstrap ADR

### PR DESCRIPTION
**Pull request by abutterworth**
The gist: We will include react-bootstrap components in Paragon. Paragon will continue to contain custom-written components, but for any component that exists in react-bootstrap we will use the react-bootstrap version instead of building our own.

### CODE REVIEW

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+5. Usage of Reactstrap
```
Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:1]: Reactstrap or react-bootstrap? 😉

Comment (abutterworth) [reply docs/decisions/0009-usage-of-react-bootstrap.rst:1]: oh god

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+Context
+-------
+
+Paragon is built on top of the Bootstrap CSS Framework. The Paragon React component library aims to mirror Bootstrap's HTML and Javascript components. To date we have not expanded the Paragon React component library to include all Bootstrap components.
+
+ADR 5 Made the case for adoption of Reactstrap. ADR 8 abandoned the idea because adopting Reactstrap "comes at the cost of abstracting away raw html and introducing yet another API to learn and remember." Since then, we have added very few homegrown components. Designers and engineers are often needing components that exist in Bootstrap but not built as Paragon components.
+
+By not building these components ourselves and not adopting third-party react components, we are wasting time and generating confusion in the edX organization.
```
Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:18]: nit: capitalize React

Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:18]: > we are wasting time and generating confusion in the edX organization.

might recommend explaining why this is the case here with a concrete example.

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+--------
+
+We will include react-bootstrap components in Paragon. Paragon will continue to contain custom-written components, but for any component that exists in react-bootstrap we will use the react-bootstrap version instead of building our own. This will provide us comprehensive coverage of Bootstrap features in react and allow us to focus on adding value by creating components outside the scope of Bootstrap.
+
+
+This will be accomplished by:
+
+1. React-bootstrap will be included as a dependency of this project. We will pin to a minor version. (Right now 1.0.x).
+
+2. Components will be passed-through from react-bootstrap by adding shell components in Paragon. (e.g. Modal.js in Paragon will simply include Modal from react-bootstrap and re-export it). This will provide us flexibility to swap the implementation in the future or enforce a11y rules if needed.
```
Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:33]: 🔥

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+3. Documentation for components included from react-bootstrap will:
+
+  - Include an interactive demo of the component in use, similar to today's components.
+  - The API will not be documented on the Paragon doc-site for now. (see next bullet)
+  - Include links to the component doc on react-bootstrap documentation site. We do this, aware that some day in the future react-bootstrap will upgrade to Bootstrap 5 and change the doc site. They did this for the switch from Bootstrap 3 to Bootstrap 4. The Bootstrap 3 doc site moved urls, but is still on the web. When that happens we will update our documentation to the new location.
+
+Other notes:
+
+- As we build new components outside the scope of Bootstrap, we will likely want to use other components to compose larger ones. For example, we may use Buttons and Cards in a CourseCard component. When we do this we must import the components from Paragon's shell components and not react-bootstrap directly. Importing from local Paragon component shells and not react-bootstrap will keep dependencies on react-bootstrap components in a single place, making future architectural changes easier.
```
Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:43]: 💯

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+- Collapsible
+- Hyperlink
+- Icon
+- MailtoLink
+- Responsive
+- SearchField
+- StatefulButton
+- ValidationFormGroup
+- TransitionReplace
```
Comment (adamstankiewicz) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:63]: Should `Pagination` be included here? react-bootstrap's `Pagination` is presentation only, whereas ours is functional. That said, the UI elements of Paragon's `Pagination` could be replaced by react-bootstrap's.

Comment (abutterworth) [reply docs/decisions/0009-usage-of-react-bootstrap.rst:62]: I think our Pagination should become ResponsivePagination and use react-bootstrap under the hood. For no let's say it stays. I'll add it to the list.

File: docs/decisions/0009-usage-of-react-bootstrap.rst (hunk 3 above)
Comment (muselesscreator) [reply docs/decisions/0009-usage-of-react-bootstrap.rst:32]: :+1:

File: docs/decisions/0009-usage-of-react-bootstrap.rst (hunk 2 above)
Comment (abutterworth) [reply docs/decisions/0009-usage-of-react-bootstrap.rst]: Added examples

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
+5. Usage of React-Bootstrap
+---------------------------
+
+
+Status
+------
+
+Adopted
```
Comment (nasthagiri) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:8]: `Adopted` -> `Accepted`

File: docs/decisions/0009-usage-of-react-bootstrap.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
+5. Usage of React-Bootstrap
```
Comment (nasthagiri) [inline docs/decisions/0009-usage-of-react-bootstrap.rst:1]: I am overall supportive of this decision.
If this ADR is accepted, as part of this PR, please update the `Status` of ADR 0008 to `Superseded by ADR-0009`.

Comment (abutterworth) [reply docs/decisions/0009-usage-of-react-bootstrap.rst:1]: Will do.

**Comment by davidjoy** [comment]
It goes without saying, but this does have the effect of tying us closer to Bootstrap and further from any sort of CSS-in-JS solution.  @abutterworth, do you think CSS-in-JS is still an important option for us?  It was going to be a heavy lift if we decided to do it, but now it'll be even heavier.  For those following along, we'd discussed CSS-in-JS as being a useful way of isolating our CSS by co-locating it with components, reducing our CSS footprint, and potentially easing theming by adopting something like [emotion-theming](https://emotion.sh/docs/theming).  

This is potentially just fine, but I wanted to bring it up for the sake of completion.

**Comment by AlasdairSwan** [comment]
My only concern with this approach is related to a11y. Are there plans to audit the ```react-bootstrap``` components and identify/fix any a11y gaps?

**Comment by abutterworth** [comment]
@davidjoy I think this may actually help us decouple from Bootstrap. Today, many Bootstrap components are written as html in our applications – this change would reduce that by pulling those components into Paragon. These new shim components we create to pass through react-bootstrap components can then be the single point to swap technology for any future innovation that takes us beyond Bootstrap.

I think CSS-in-JS is a good future for us. If we choose to adopt it we should do it iteratively (and therefore in a way that is compatible with Bootstrap being present on the page). I imagine a future where we adopt new technology by replacing our react-bootstrap shims 1 by 1 (removing sections of bootstrap scss along the way).

**Comment by abutterworth** [comment]
@AlasdairSwan I'll reach out to @wittjeff to request reviews of these components as we pull them in.

**Comment by abutterworth** [comment]
@wittjeff has done a quick a11y sniff test on the react-bootstrap components here: https://docs.google.com/spreadsheets/d/1FR42auyyFPfOLenlcj6OLuYZ7NU0aBTZvYIM58iSElQ/edit#gid=0

I'll update this PR to reflect the following plan:
- For the 19 components that pass our initial audit we will pull them in
- For the 8 that do not we will make upstream changes. Or, if they are needed urgently we will enforce a11y requirements in the shim component.

**Pull request merged**